<a href="https://colab.research.google.com/github/Paiz-lab/Data_Kantin/blob/main/Analisis_Data_Kantin.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


Tugas Analisis 1:

Jika kita mencoba [5000, 7000, 3000] * 0.9 pada list biasa, Python akan menghasilkan TypeError / Error karena list Python tidak mendukung operasi matematika langsung (vectorized operations) dengan angka desimal (float).

Mengapa berbeda? Array NumPy dirancang khusus untuk komputasi matematika, sehingga perkalian langsung diterapkan ke setiap elemen di dalamnya (element-wise). Sedangkan pada list Python biasa, operator * hanya bisa digunakan dengan angka bulat (int) untuk mengulang isi list (duplikasi).

In [2]:
import pandas as pd

data_kantin = {
    'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
    'harga': [12000, 4000, 10000, 4000, 8000],
    'terjual': [23, 40, None, 35, 18]
}

df_latihan = pd.DataFrame(data_kantin)
print(df_latihan)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


Tugas Analisis 2:

Kolom yang memiliki data kosong (None/NaN) adalah kolom menu (baris index ke-4) dan kolom terjual (baris index ke-2).

Risikonya: Jika data langsung dianalisis tanpa dibersihkan, perhitungan statistik (seperti rata-rata atau total penjualan) bisa menjadi tidak akurat atau menghasilkan angka NaN. Selain itu, saat membuat visualisasi data/grafik, data kosong bisa menyebabkan error atau informasi yang salah.

In [3]:
df = pd.read_csv('data_kantin.csv')

print("--- 5 Baris Pertama ---")
print(df.head())

print("\n--- Info Data ---")
df.info()

print("\n--- Ringkasan Statistik ---")
print(df.describe())

print("\n--- Ukuran Data ---")
print(df.shape)

--- 5 Baris Pertama ---
      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0

--- Info Data ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB

--- Ringkasan Statistik ---
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.0000

Tugas Analisis 3:

Kolom yang jumlah non-null-nya lebih sedikit dari total baris (110 baris) adalah kolom menu (hanya 107 non-null) dan kolom terjual (hanya 104 non-null).

Artinya: Terdapat missing values (data kosong/hilang) pada kedua kolom tersebut. Kolom menu memiliki 3 data kosong, dan kolom terjual memiliki 6 data kosong.

In [4]:
print(df.isnull().sum()) # jumlah data kosong tiap kolom
df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


Tugas Analisis 4:

Kolom terjual diisi dengan fillna(0) karena merupakan data numerik (jumlah barang). Asumsinya, jika tidak dicatat/kosong, berarti tidak ada barang yang terjual pada hari itu (0 unit), sehingga data transaksi lain pada baris tersebut masih bisa dianalisis.

Kolom menu dihapus dengan dropna() karena kolom menu adalah identitas utama dari transaksi (kategori/nama produk). Jika nama produknya kosong, data transaksi tersebut menjadi tidak valid dan tidak bisa diketahui barang apa yang dijual.

In [5]:
print("Jumlah baris duplikat:", df.duplicated().sum())
df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int)
print(df.dtypes)

Jumlah baris duplikat: 5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


Tugas Analisis 5:

Jumlah baris dataset: Berkurang dari 107 baris menjadi 102 baris (terhapus 5 baris data duplikat).

Pentingnya memastikan tipe data (dtypes) sudah benar:

Memastikan operasi matematika (seperti perkalian harga dan jumlah terjual) berjalan dengan benar dan akurat.

Mencegah terjadinya error saat pengolahan data atau pembuatan visualisasi (misalnya tipe data teks/string tidak bisa dijumlahkan).

Efisiensi penggunaan memori pada notebook.

In [6]:
laris = df[df['terjual'] > 20]
urut = df.sort_values(by='terjual', ascending=False)

df['total_pendapatan'] = df['harga'] * df['terjual']

ringkasan = df.groupby('menu')['total_pendapatan'].sum()
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


Tugas Analisis 6:

Menu dengan total pendapatan tertinggi: Nasi Goreng dengan total pendapatan mencapai Rp3.924.000.

Manfaat informasi ini untuk pengambilan keputusan di kantin sekolah:

Manajemen Stok/Bahan Baku: Pengelola kantin dapat menambah persediaan stok bahan baku untuk Nasi Goreng agar tidak kehabisan di jam istirahat.

Penentuan Prioritas Menu: Fokus menjaga kualitas dan pelayanan pada menu-menu penyumbang omset terbesar (seperti Nasi Goreng dan Mie Ayam).

Strategi Promosi/Bundling: Menu yang kurang menghasilkan pendapatan bisa dibuatkan paket hemat (bundling) bersama menu terlaris untuk meningkatkan penjualannya.

# **PROYEK AKHIR EDA**


In [7]:
import pandas as pd
df = pd.read_csv('data_kantin.csv')
df.head()
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB


,harga,terjual
count,110.000000,104.000000
mean,6336.363636,22.923077
std,3520.001232,10.831178
min,2000.000000,8.000000
25%,3000.000000,13.000000
50%,5000.000000,20.000000
75%,10000.000000,33.000000
max,12000.000000,45.000000


In [8]:
# 1. Hapus duplikat
df = df.drop_duplicates()

# 2. Isi missing value (harus dieksekusi agar non-null count jadi sama)
df['menu'] = df['menu'].fillna('Tidak Diketahui')
df['terjual'] = df['terjual'].fillna(0)
# 3. Ubah tipe data
df['tanggal'] = pd.to_datetime(df['tanggal'])
df['terjual'] = df['terjual'].astype(int)

# 4. Cek hasil akhir
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 105 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   tanggal   105 non-null    datetime64[ns]
 1   menu      105 non-null    object        
 2   kategori  105 non-null    object        
 3   harga     105 non-null    int64         
 4   terjual   105 non-null    int64         
dtypes: datetime64[ns](1), int64(2), object(2)
memory usage: 4.9+ KB


In [9]:
# 1. Kolom Turunan: Menghitung total omset/pendapatan per transaksi
df['total_pendapatan'] = df['harga'] * df['terjual']

# 2. Filtering: Mengambil transaksi dengan pendapatan di atas 50.000
df_filtered = df[df['total_pendapatan'] > 50000]

# 3. Sorting: Mengurutkan dari transaksi dengan pendapatan tertinggi
df_sorted = df.sort_values(by='total_pendapatan', ascending=False)

# 4. Groupby / Agregasi: Total pendapatan & total unit terjual berdasarkan Kategori
df_grouped = df.groupby('kategori')[['terjual', 'total_pendapatan']].sum().reset_index()

# Tampilkan hasil pengelompokan
df_grouped


,kategori,terjual,total_pendapatan
0,Makanan,762,8018000
1,Minuman,1033,4260000
2,Snack,486,972000


In [10]:
df.to_csv('dataset_bersih.csv', index=False)

# Lembar Perencanaan Proyek Kelompok (Project Charter)

* **Nama Anggota Kelompok**:
  1. Rafa Mahardika Setiawan
  2. Muhammad Faiz Najwan
* **Dataset yang Dipilih**: Data Penjualan Kantin/Toko (`data_kantin.csv`)
* **Pertanyaan Analisis Awal**:
  1. Menu apa yang paling laris dan menghasilkan pendapatan terbanyak?
  2. Kategori mana (Makanan, Minuman, Snack) yang memiliki total penjualan tertinggi?
  3. Berapa rerata penjualan harian untuk tiap produk?
* **Dugaan Masalah Kualitas Data**:
  * Missing value pada kolom `menu` dan `terjual`
  * Data duplikat (terdapat 5 baris duplikat)
  * Tipe data `tanggal` masih bernilai `object` (harus diubah ke `datetime`)
* **Rencana Teknik Pembersihan**:
  * Menghapus baris duplikat (`drop_duplicates()`)
  * Mengisi atau menghapus missing value (`dropna()` untuk menu yang tidak bisa ditebak, atau `fillna()` sesuai median/modus)
  * Mengonversi tipe data `tanggal` ke `datetime` dan `terjual` ke `int`
* **Rencana Manipulasi Data**:
  * Filter: `df[df['kategori'] == 'Makanan']`
  * Sort: `df.sort_values(by='terjual', ascending=False)`
  * Kolom Turunan: `total_pendapatan = harga * terjual`
  * Groupby: `df.groupby('kategori')['total_pendapatan'].sum()`
* **Jadwal Kerja**:
  * P3 (Loading & Inspection): Hari 1
  * P4 (Cleaning): Hari 2
  * P5 (Manipulation): Hari 3
  * P6 (Uji & Presentasi): Hari 4
* **Pembagian Peran**:
  * Anggota 1: Data Cleaning & Loading
  * Anggota 2: Data Manipulation & Summary